File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnAssetFetch.cs

namespace GracenoteSDK {

/**
* <b>Experimental</b>: Fetch asset raw data.
* The raw data is fetched during object construction
* and may result in lengthy network access if accessing
* from online.
*/
public class GnAssetFetch : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnAssetFetch(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnAssetFetch obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnAssetFetch() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnAssetFetch(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

      public byte[] DataBuffer
      {
        get
        {
          byte[] dataBytes = new byte[Size];
          System.Runtime.InteropServices.Marshal.Copy((System.IntPtr)Data, dataBytes, 0, (int)Size);
          return dataBytes;
        }
      }
    
/**
* Perform the fetch of the data from the provided URL.
* This method can fetch data from any valid URL. This class must
* be used to fetch data from a GNSDK local URL.
* Construction may result in lengthy network access if the URL is
* online and the target data is large.
*/
  public GnAssetFetch(GnUser user, string url, GnStatusEventsDelegate pEventHandler) : this(gnsdk_csharp_marshalPINVOKE.new_GnAssetFetch__SWIG_0(GnUser.getCPtr(user), url, GnStatusEventsDelegate.getCPtr(pEventHandler)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnAssetFetch(GnUser user, string url) : this(gnsdk_csharp_marshalPINVOKE.new_GnAssetFetch__SWIG_1(GnUser.getCPtr(user), url), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnAssetFetch(GnUser user, string url, GnLookupMode lookupMode, GnStatusEventsDelegate pEventHandler) : this(gnsdk_csharp_marshalPINVOKE.new_GnAssetFetch__SWIG_2(GnUser.getCPtr(user), url, (int)lookupMode, GnStatusEventsDelegate.getCPtr(pEventHandler)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnAssetFetch(GnUser user, string url, GnLookupMode lookupMode) : this(gnsdk_csharp_marshalPINVOKE.new_GnAssetFetch__SWIG_3(GnUser.getCPtr(user), url, (int)lookupMode), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public string Type() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnAssetFetch_Type(swigCPtr);
    return ret;
  }

/**
* Get the event handler provided on construction
* @return Event handler
*/
  public GnStatusEventsDelegate EventHandler() {
    global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnAssetFetch_EventHandler(swigCPtr);
    GnStatusEventsDelegate ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnStatusEventsDelegate(cPtr, false);
    return ret;
  }

/**
* Raw data size in bytes
*/
  public uint Size {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnAssetFetch_Size_get(swigCPtr);
      return ret;
    } 
  }

  public System.IntPtr Data {get{System.IntPtr res = gnsdk_csharp_marshalPINVOKE.GnAssetFetch_Data_get(swigCPtr); ; return res;}
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnAssetFetch.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b>: Fetch asset raw data. 
* The raw data is fetched during object construction 
* and may result in lengthy network access if accessing 
* from online. 
*/ 
 
public class GnAssetFetch {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnAssetFetch(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnAssetFetch obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnAssetFetch(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

	private IGnStatusEvents pEventHandler;
	private IGnStatusEventsProxyU eventHandlerProxy;

/** 
* Perform the fetch of the data from the provided URL. 
* This method can fetch data from any valid URL. This class must 
* be used to fetch data from a GNSDK local URL. 
* Construction may result in lengthy network access if the URL is 
* online and the target data is large. 
*/ 
 
  public GnAssetFetch(GnUser user, String url, IGnStatusEvents pEventHandler) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
	swigCPtr = gnsdk_javaJNI.new_GnAssetFetch__SWIG_0(GnUser.getCPtr(user), user, url, (eventHandlerProxy==null)?0:IGnStatusEventsProxyL.getCPtr(eventHandlerProxy), eventHandlerProxy);
}

  public GnAssetFetch(GnUser user, String url) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
	swigCPtr = gnsdk_javaJNI.new_GnAssetFetch__SWIG_1(GnUser.getCPtr(user), user, url);
}

  public GnAssetFetch(GnUser user, String url, GnLookupMode lookupMode, IGnStatusEvents pEventHandler) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
	swigCPtr = gnsdk_javaJNI.new_GnAssetFetch__SWIG_2(GnUser.getCPtr(user), user, url, lookupMode.swigValue(), (eventHandlerProxy==null)?0:IGnStatusEventsProxyL.getCPtr(eventHandlerProxy), eventHandlerProxy);
}

  public GnAssetFetch(GnUser user, String url, GnLookupMode lookupMode) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
	swigCPtr = gnsdk_javaJNI.new_GnAssetFetch__SWIG_3(GnUser.getCPtr(user), user, url, lookupMode.swigValue());
}

/** 
* Raw data buffer 
*/ 
 
  public byte[] data() {
	return gnsdk_javaJNI.GnAssetFetch_data(swigCPtr, this);
}

/** 
* Raw data size in bytes 
*/ 
 
  public long size() {
    return gnsdk_javaJNI.GnAssetFetch_size(swigCPtr, this);
  }

  public String type() {
    return gnsdk_javaJNI.GnAssetFetch_type(swigCPtr, this);
  }

/** 
* Get the event handler provided on construction 
* @return Event handler 
*/ 
 
  public IGnStatusEvents eventHandler() {
	return pEventHandler;
}

}
